test/gtest/conv_2d_implicit_gemm.cpp

```diff
@@ -0,0 +1,241 @@
+        tokens.push_back(*begin++);
+}
+
+class Conv2dHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+class Conv2dInt8 : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+class Conv2dBFloat16 : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+class Conv2dFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+void Run2dDriver(Precision prec)
+{
+
+    std::vector<TestCase> params;
+    switch(prec)
+    {
+    case Precision::Float: params = Conv2dFloat::GetParam(); break;
+    case Precision::Half: params = Conv2dHalf::GetParam(); break;
+    case Precision::Int8: params = Conv2dInt8::GetParam(); break;
+    case Precision::BFloat16: params = Conv2dBFloat16::GetParam(); break;
+    default: params = Conv2dFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        for(std::string const& str : tokens)
+            ptrs.push_back(str.data());
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        EXPECT_FALSE(capture.find("Perf Db: record not found") != std::string::npos);
+    }
+};
+
+TEST_P(Conv2dFloat, FloatTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--float")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::Float);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+TEST_P(Conv2dHalf, HalfTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--half")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::Half);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+TEST_P(Conv2dInt8, Int8Test)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--int8")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::Int8);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+TEST_P(Conv2dBFloat16, BFloat16Test)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--bfloat16")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::BFloat16);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+std::vector<TestCase> GetTestCases(const std::string& precision)
+{
+    const std::vector<TestCase> test_cases = {
+        // clang-format off
+    std::make_tuple<std::vector<std::string>, std::string>(
+        {"MIOPEN_DEBUG_AMD_WINOGRAD_RXS_F3X2=0", "MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_WRW_V4R1=0"},
```

Comment:
these environment variables were used as WA for issue #874, we should be able to disable all this WA without changing the test code.

Reply:
> these environment variables were used as WA for issue #874, we should be able to disable all this WA without changing the test code.

@shurale-nkn , I'm working on the PR now. Do you mean MIOPEN_DEBUG_AMD_WINOGRAD_RXS_F3X2 and MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_WRW_V4R1 are dated and can be removed?

Reply:
No! This is workaround for this issue. Issue is still open. You must save the reference to this issue in your code so that it can be quickly found and deleted later. Also, the person who will fix this error should be able to quickly disable this WA in order to test MIOpen without it.

Reply:
> Also, the person who will fix this error should be able to quickly disable this WA in order to test MIOpen without it.

@shurale-nkn , brought back GetEnvVars() which will enable/disable the WA with the env vars e.g. MIOPEN_DEBUG_AMD_WINOGRAD_RXS_F3X2, MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_WRW_V4R1

Reply:
@shurale-nkn , this WA has been lift. Here's the PR
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2179

Reply:
[Resolved]

Reply:
[Notice] W/As for #874 and #1008 finally removed by #2220